test/mpi/init/slurm_nodelist_parse.c

```diff
@@ -0,0 +1,326 @@
+#include <stdio.h>
```

Comment:
@hzhou I have added this quickly to see if it can be part of the test suite. `test/mpi/init` might not be the best place for this test however. BTW adding this test I have found another bug :)